src/hotspot/share/prims/jvmtiRawMonitor.hpp

```diff
@@ -111,8 +111,8 @@ class JvmtiRawMonitor : public CHeapObj<mtSynchronizer>  {
   };
 
   // Non-aborting operator new
-  void* operator new(size_t size) {
-    return CHeapObj::operator new(size, std::nothrow);
+  void* operator new(size_t size, const std::nothrow_t&  nothrow_constant) throw() {
```

Comment:
Hm, now I'm wondering why isn't an `operator delete` to go with this?  Or are these objects
never deleted?  Otherwise I'd have thought we'd get the same mismatched new/delete warning
you encountered elsewhere.  If they're never supposed to be deleted, then giving `operator delete`
a deleted definition here seems appropriate, to prevent accidentally calling the CHeapObj function.

Reply:
This `operator new` just calls the `CHeapObj::operator new` with nothrow argument. So changing the caller will call the right one in `CHeapObj`. This object is deleted in  
https://github.com/openjdk/jdk/blob/c738c8ea3e9fda87abb03acb599a2433a344db09/src/hotspot/share/prims/jvmtiEnv.cpp#L3699
and this will call the `CHeapObj::operator delete` which is the right one. So this `operator new` is not needed since I changed the caller.

Reply:
A possible reason for keeping this `operator new` is to force the use of null return for oom for this class.
If it's removed then we have the option of (perhaps unintentionally) using the terminating allocator.
That doesn't seem like a _strong_ reason to keep it, but someone more familiar with jvmti stuff might
want to weigh in.  If it is kept, then I think it should have a corresponding `operator delete`, else it at
least looks odd.

Reply:
JVMTI does not abort on OOM it reports an error, so we definitely do not want a terminating allocator!
```
jvmtiError
JvmtiEnv::CreateRawMonitor(const char* name, jrawMonitorID* monitor_ptr) {
  JvmtiRawMonitor* rmonitor = new JvmtiRawMonitor(name);
  NULL_CHECK(rmonitor, JVMTI_ERROR_OUT_OF_MEMORY);
```

Reply:
The new operator is removed.

Reply:
We decide this at the call site though by adding the nothrow parameter.  Adding an overloaded operator new without a nothrow parameter that we're not supposed to call seems very marginally useful.  ie not useful.